Solar-Wrapper/src/TrackablesSolARImpl.cpp

```diff
@@ -0,0 +1,287 @@
+
+        //trackable UUID
+        std::string id = xpcf::uuids::to_string(trackable.getID());
+        ret.setUUID(id);
+
+        //creator UUID
+        std::string creatorUid = xpcf::uuids::to_string(trackable.getCreatorID());
+        ret.setCreatorUUID(creatorUid);
+
+        //Trackable type
+        std::string type = resolveTrackableType(trackable.getType());
+        ret.setTrackableType(type);
+
+        //EncodingInfo struct
+        EncodingInformationStructure encodingInfo;
+        EncodingInfo storageEncodingInfo(trackable.getEncodingInfo());
+        encodingInfo.setDataFormat(storageEncodingInfo.getDataFormat());
+        encodingInfo.setVersion(storageEncodingInfo.getVersion());
+        ret.setTrackableEncodingInformation(encodingInfo);
+
+        //Payload
+        std::vector<std::byte> storagePayload = trackable.getPayload();
+        std::string payload(reinterpret_cast<const char *>(&storagePayload[0]), storagePayload.size());
+        ret.setTrackablePayload(payload);
+
+        //Transform3Df (localCRS)
+        Transform3Df transform3d = trackable.getLocalCrs();
+        std::vector<float> localCRS;
+        size_t nCols = transform3d.cols();
+        for (size_t i = 0; i < nCols; ++i)
+           for (size_t j = 0; j < nCols; ++j)
+           {
+                localCRS.push_back(transform3d(j, i));
+           }
+        ret.setLocalCRS(localCRS);
+
+        //Unit system
+        UnitSystem unit = resolveUnitSystem(trackable.getUnitSystem());
+        ret.setUnit(unit);
+
+        //Dimension (scale)
+        Vector3d dimension = trackable.getSize();
+        std::vector<double> vector(3);
+        for(int i = 0; i < 3; i++){
+            vector[i] = dimension[0,i];
+        }
+        ret.setTrackableSize(vector);
+
+        //keyvalue taglist (multimap to map<string,vector<string>>)
+        std::map<std::string, std::vector<std::string>> tagList;
+        auto storageMap = trackable.getTags();
+        for (auto itr = storageMap.begin() ; itr != storageMap.end(); itr++){
+                if (tagList.count(itr->first) != 0){
+                    std::vector<std::string>& vector = tagList.at(itr->first);
+                    vector.push_back(itr->second);
+                }else {
+                    std::vector<std::string> vector;
+                    vector.push_back(itr->second);
+                    tagList.insert(std::pair<std::string, std::vector<std::string>>(itr->first, vector));
+                }
+        }
+        ret.setKeyvalueTags(tagList);
+
+        return ret;
+    }
+
+    std::vector<std::byte> TrackablesSolARImpl::toBytes(std::string const& s)
+    {
+        std::vector<std::byte> bytes;
+        bytes.reserve(std::size(s));
+        std::transform(std::begin(s), std::end(s), std::back_inserter(bytes), [](char c){
+            return std::byte(c);
+        });
+
+        return bytes;
+    }
+
+    void TrackablesSolARImpl::init(){
+        TrackablesApi::init();
+        try {
+        }
```

Comment:
If there's nothing in this block, there's a fat chance something will ever get caught.